### DISCUSSION TITLE
Push through an image that will be suitable for practical use

### DISCUSSION THREAD

**Discussion by exotsk** [HASwitchPlate/openHASP#439] [Q&A]
I bought sunton32-capacity by using it as an "exit-door conrtoller." So that when leaving, you can check the parameters of the house, turn on the alarm, and see who is the caller outside the door (one of them is far away). I wanted when I pressed the call, a small static image of 10-30 kb, to be sent to the screen, and it could be viewed. but the maximum I got was 140pixels. as in the picture below. (thank you so much for helping @htvekov, for explaining in detail what to do) I have another fee, the project has a different purpose, does not have buttons but can accept an image on almost the entire screen. Is it possible to somehow develop this direction of the project? Or give some practical advice if need psram whether it is possible to install it with an additional module to the board.

![image](https://user-images.githubusercontent.com/54969124/222427943-e23a8ba0-266b-41b7-9f03-94be48fd959f.png)
![image](https://user-images.githubusercontent.com/54969124/222428355-21fdffcd-0ebe-4424-bc17-27d30b369b85.png)

**Comment by htvekov** [comment]
I'm guessing a bit here, as you need to provide more details in order for me to understand the exact goal 😉

In general. Yes, you'll need PSram in order to process images - besides small thumbnails/objects.
What Sunton device (model no.) are you using for this ?

Regarding image size. openHASP custom component didn't until recently support upscaling of images - only downscaling.
So if your not using latest custom component from main branch (dev.) then upscaling is not supported.
If your orignal image has a width of 140 pixels, you can't make that image larger without CC from main branch.

In order to upscale images, you'll have to set the new `fitscreen` parameter (example below)

```yaml
- service: openhasp.push_image
          data:
            obj: p2b40
            image: "http://{{states('sensor.local_ip')}}:8123{{state_attr(state_attr('group.sonos_all', 'entity_id')[0], 'entity_picture')}}"
            fitscreen: 1
            width: >
              {% if w == 800 %}
                355
              {% else %}
                {{ w }}
              {% endif %}
            height: >
              {% if w == 800 %}
                355
              {% else %}
                {{ w }}
              {% endif %}
          target:
            entity_id: >
              {{ repeat.item }}
```
Note that when upscaling there's **_no_** check on the image aspect ratio !

**Reply by gudszent** [reply]
Am I correct if I say the base image came from frigate? I try to reach almost a same thing(I don have HA, just openHAB), but that give jpg and that not works for me...

**Comment by exotsk** [comment]
thank you very much for the answer.
I use the 3.5inch_ESP32-3248S035, the latest firmware dev version and the main version of the HA component
The firmware was compiled through the GitPod, only unannotated; user_setups/esp32/*.ini and  ; esp32-3248s035c_4MB

![image](https://user-images.githubusercontent.com/54969124/222887318-d2a2b129-f0dc-4800-a055-d9cd3269ce75.png)

and tried to send photos of different sizes (640x360) and formats (jpg png) through HA service.
I used the fitscreen key but then the picture was not displayed at all. not if the width or height is greater than 140 pixels.
This is clearly due to the lack of memory for the picture operation. although jpg itself (640x360) weighs about 15 kBytes


using the option fitscreeen, nothing changes, no image on screen.
![image](https://user-images.githubusercontent.com/54969124/222887553-6c13562f-7019-43d7-82e1-404ccf31c91e.png)

**Reply by htvekov** [reply]
Well, without PSram on the device, you can't push images much larger than thumbnails - nothing to do about that.
I would advise you to buy a higher spec device with esp32-s3 mcu, at least 8MB flash and PSram (2 MB will do. But most devices today has 8MB) 

Doesn't really matter if the original, compressed image is only a small file at appx. 15kb. as Custom Component converts all images to binary files

commit 1427244 is ok and latest CC version. This version includes the `fitscreen` parameter

**Comment by exotsk** [comment]
It is clear that you will have to order a third screen:) is such a version suitable for my purposes? will the full screen picture be able to process?
![image](https://user-images.githubusercontent.com/54969124/222905804-e7235d3e-a41c-4054-a333-5b4f7307f8d3.png)

**Reply by htvekov** [reply]
I don't own that particular Sunton device. Currently I've only one Sunton device - the 7" version
So I can't recommend one specific Sunton device for you to buy.
Just be aware that some of the device models have numerous variants. With / without touch,  Resistive / Capasive touch, TFT / IPS display.
As Suntons device prices doesn't differ that much, I would go for a device with a high resolution IPS display type, capasitive touch, ESP32-S3 mcu, 16MB flash and 8MB PSram.

I believe the largest device Sunton currently has released that fulfills all these requirements is the 5" version ([ESP32-8048S050](https://www.aliexpress.com/item/1005004952694042.html))
The Sunton ESP32-4827S043 4,3" version with IPS display and capasitive touch, is only a few dollars cheaper than the 5" version, but has significantly lower display resolution. So I would definitely go for the the 5" version, if the larger size is not an issue 🙂

**Comment by exotsk** [comment]
i suppose on the picture is  4.3inch_ESP32-8048S043 IPS capacty. ESP32-8048S050 just has low dpi ?  it's both support by openHASP?
here is only 4.3 version.  i don't need 4 sunton plate. =)


    ; esp32-2432s028r_4MB
    ; esp32-3248s035c_4MB
    ; esp32-3248s035c_4MB_lovyan
    ; esp32-3248s035r_4MB
    ; esp32-s3-4827s043c_16MB
    ; esp32-s3-4827s043r_16MB
    ; esp32-s3-8048s043c_16MB

**Reply by htvekov** [reply]
Hmm... I guess that the predefined list is not updated.
The 5" version is available though
@fvanroie Can you confirm ?

![image](https://user-images.githubusercontent.com/45469813/222911843-daea24bf-ac87-408d-a350-64a6b95f6f35.png)

**Reply by fvanroie** [reply]
That 'predefined list' are examples only. Use one of the valid names instead.
Please file a Pull Request for updates to the list.

**Reply by htvekov** [reply]
> That 'predefined list' are examples only. Use one of the valid names instead. Please file a Pull Request for updates to the list.

Noted 👍 I thought that envs list was regularly maintained and not just an example list.
Too many devices these days 😃 I'll find some time to do that PR for the esp32_s3 devices

@exotsk Yes, the 5" Sunton device is supported as well

**Comment by exotsk** [comment]
@htvekov =) Thanks .... it`s perfecto 
![image](https://user-images.githubusercontent.com/54969124/229788309-a288d546-1232-4586-954d-eb178d9c467c.png)

**Reply by supermarcie** [reply]
can you provide me with the code on the display and the code in Home assistant please.
Because I don't get it working
